### ISSUE TITLE
Keep a running total of disk space used by each form and user

### ISSUE THREAD

**Issue by jnm** [kobotoolbox/kobocat#517]
We already count submissions:
https://github.com/kobotoolbox/kobocat/blob/568c7aea243fb73a68e9fc6dca8753146e02935d/onadata/libs/utils/logger_tools.py#L274-L275

We should also note the size of each incoming attachment and add that to a total for (1) the form and (2) for the user owning the form. This will help us identify power users :grinning: and people who are overwhelming our infrastructure :grimacing:

**Comment by jnm** [comment]
I did #653, but it misses the mark because it doesn't include storage used by thumbnails (see below for how they're created).
https://github.com/kobotoolbox/kobocat/blob/6eef7d51ce5fb3f071b61f1767b486752e59fa15/onadata/libs/utils/image_tools.py#L69-L92

Every uploaded attachment ends up being stored up to 4 times:
https://github.com/kobotoolbox/kobocat/blob/6eef7d51ce5fb3f071b61f1767b486752e59fa15/onadata/settings/common.py#L421-L425

These thumbnails aren't created automatically upon upload, but rather as they're requested for display by various parts of the front end or directly by power users. However, once they're created, they're stored indefinitely :grimacing:

@tinok, should we count thumbnails as part of the total storage occupied by a user? Either way, we might want to purge these thumbnails over time—what do you think, @noliveleger?

**Comment by tinok** [comment]
@jnm My immediate feeling is that the cost of thumbnails is small compared to the effort to properly count them. One solution could be to multiply each (image?) attachment size by 1.018. Alternatively, we could purge attachments after x days since they can be re-created by the FE at any time. 

To confirm, do only image files get thumbnails or all attachments?

**Comment by noliveleger** [comment]
@jnm I would agree to purge. As Tino said, they can be created on the fly if needed.

**Comment by jnm** [comment]
Right, just image files.

I guess with S3 we can have files expire automatically(?) I'm not sure how we'd deal with local storage without adding a periodic task—not that it's out of the question. It might also not matter so much since installations with local storage tend to be smaller.

**Comment by tinok** [comment]
I thought "expiration" refers just to how long they can be accessed with a unique URL / token? But I know very little about S3.

@noliveleger are there important thresholds in terms of intelligent tiering that would impact re-creating thumbnails? I.e., could it be more expensive to recreate thumbnails for the gallery after 60 days than to just keep them around?

**Comment by noliveleger** [comment]
@tinok, good question. 

Intelligent tiering works this way: 
- first 30 days: S3 std prices ($0.023 / GB)
- rest of the life: S3 Infrequent access ($0.0125 / GB)
- monitoring (to switch files from one class to another): $0.0025 per 1,000 objects

Each time a file is accessed, it is moved to `standard` class. So its cost is the same as a new file. The difference is the monitoring cost during file life. 
So I don't see any cons to delete thumbnails after X days.

**Comment by tinok** [comment]
Sounds good. What time frame should we use for purging thumbnails then? 90 days?
